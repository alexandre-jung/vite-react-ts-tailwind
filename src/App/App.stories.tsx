import { ComponentStory, ComponentMeta } from '@storybook/react';
import AppComponent from '.';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Application/App',
  component: AppComponent,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as ComponentMeta<typeof AppComponent>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof AppComponent> = () => <AppComponent />;

export const App = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
App.args = {};
