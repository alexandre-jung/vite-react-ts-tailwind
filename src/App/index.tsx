import Counter from '@components/Counter';
import { ReactComponent as ReactLogo } from '@assets/img/react-logo.svg';
import styles from './styles.module.scss';
// import { useEffect, useState } from 'react';
// import Expenses from '@routes/expenses';
// import Home from '@routes/home';
// import Invoice from '@routes/invoice';
// import Invoices from '@routes/invoices';
// import { NavLink, Route, Routes } from 'react-router-dom';

export default function App() {
  // const [data, setData] = useState<{ [key: string]: string }[]>();
  // const [isLoading, setIsLoading] = useState(false);

  // useEffect(() => {
  //   async function fetchData() {
  //     try {
  //       const response = await fetch('http://localhost:3000/api/users');
  //       const data = await response.json();
  //       setData(data.data);
  //     } catch (error) {
  //       console.log(error);
  //     } finally {
  //       setIsLoading(false);
  //     }
  //   }
  //   setIsLoading(true);
  //   fetchData();
  // }, []);

  // if (isLoading) {
  //   return <div>...</div>;
  // }

  // return (
  //   <div>{data && data.map((user) => <p key={user.id}>{user.name}</p>)}</div>
  // );

  return (
    <div className={styles.App}>
      <ReactLogo width={512} height={512} className="text-sky-600 rotate m-8" />
      <h1 className={styles.title}>Vite + React template project</h1>
      <Counter />
    </div>
  );

  // <div id="app">
  //   <nav>
  //     <NavLink
  //       style={({ isActive }) => ({
  //         color: isActive ? 'red' : '',
  //       })}
  //       to="/"
  //     >
  //       home
  //     </NavLink>{' '}
  //     <NavLink
  //       style={({ isActive }) => ({
  //         color: isActive ? 'red' : '',
  //       })}
  //       to="expenses"
  //     >
  //       expenses
  //     </NavLink>{' '}
  //     <NavLink
  //       style={({ isActive }) => ({
  //         color: isActive ? 'red' : '',
  //       })}
  //       to="invoices"
  //     >
  //       invoices
  //     </NavLink>
  //   </nav>
  //   <Routes>
  //     <Route path="/" element={<Home />}>
  //       <Route path="expenses" element={<Expenses />} />
  //       <Route path="invoices" element={<Invoices />}>
  //         <Route index element={<p>Select an invoice</p>} />
  //         <Route path=":invoiceId" element={<Invoice />} />
  //       </Route>
  //       <Route path="*" element={<main>There&apos;s nothing here!</main>} />
  //     </Route>
  //   </Routes>
  // </div>
  // );
}
