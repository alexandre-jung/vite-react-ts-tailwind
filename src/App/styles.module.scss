.App {
  @apply flex flex-col justify-center items-center min-h-full;
}

.title {
  @apply text-white text-4xl mb-12;
}
