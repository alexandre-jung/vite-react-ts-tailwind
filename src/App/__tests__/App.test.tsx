import '@testing-library/jest-dom';
// import { screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App from '..';

import { render } from '@testing-library/react';

describe('App unit test', () => {
  it('should render', () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  // it('should fetch users', async () => {
  //   const { container } = render(<App />);

  //   await screen.findByText('alex');
  //   console.log(container.innerHTML);
  // });
});
