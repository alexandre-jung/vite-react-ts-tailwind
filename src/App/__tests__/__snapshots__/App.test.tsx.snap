// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App unit test should render 1`] = `
<div>
  <div
    class="stylesModule.App"
  >
    <svg
      class="text-sky-600 rotate m-8"
      height="512"
      width="512"
    >
      react-logo.svg
    </svg>
    <h1
      class="stylesModule.title"
    >
      Vite + React template project
    </h1>
    <div
      class="z-0"
    >
      <button
        class="btn rounded"
      >
        -
      </button>
      <span
        class="counter inline-block min-w-[3rem]"
      >
        0
      </span>
      <button
        class="btn rounded"
      >
        +
      </button>
    </div>
  </div>
</div>
`;
